services/src/main/java/org/apache/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -744,6 +749,8 @@ public void onComplete(Result result)
       }
       emitQueryTime(requestTimeNs, success, sqlQueryId, queryId);
 
+      AuthenticationResult authenticationResult = AuthorizationUtils.authenticationResultFromRequest(req);
```

Comment:
Since we are [attaching an authenticator even if the authenticators are null](https://github.com/apache/druid/blob/master/server/src/main/java/org/apache/druid/server/initialization/AuthenticatorMapperModule.java#L82), I'd expect the request would have an auth result every time. The QueryLifecycle also uses the same method and so if there were a failure, it would fail before the request comes back to the router.

Reply:
Yeah. I did check that, too. Just being more risk averse, I guess. I am going to approve. Feel free to merge once CI is green. Those failures are likely code coverage issues.